portal/src/graphql/portal/CreateOAuthClientScreen.tsx

```diff
@@ -0,0 +1,250 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import {
+  Callout,
+  Dialog,
+  DialogFooter,
+  DirectionalHint,
+  IconButton,
+  Label,
+  PrimaryButton,
+  Text,
+} from "@fluentui/react";
+import { useNavigate, useParams } from "react-router-dom";
+import produce from "immer";
+import deepEqual from "deep-equal";
+import { FormattedMessage } from "@oursky/react-messageformat";
+
+import ShowError from "../../ShowError";
+import ShowLoading from "../../ShowLoading";
+import ModifyOAuthClientForm from "./ModifyOAuthClientForm";
+import ButtonWithLoading from "../../ButtonWithLoading";
+import NavigationBlockerDialog from "../../NavigationBlockerDialog";
+import NavBreadcrumb, { BreadcrumbItem } from "../../NavBreadcrumb";
+import { useAppConfigQuery } from "./query/appConfigQuery";
+import { useUpdateAppConfigMutation } from "./mutations/updateAppConfigMutation";
+import { OAuthClientConfig, PortalAPIAppConfig } from "../../types";
+import { clearEmptyObject } from "../../util/misc";
+import { genRandomHexadecimalString } from "../../util/random";
+import { copyToClipboard } from "../../util/clipboard";
+
+import styles from "./CreateOAuthClientScreen.module.scss";
+
+interface CreateOAuthClientFormProps {
+  rawAppConfig: PortalAPIAppConfig;
+}
+
+interface CreateClientSuccessDialogProps {
+  visible: boolean;
+  clientId: string;
+}
+const CreateClientSuccessDialog: React.FC<CreateClientSuccessDialogProps> = function CreateClientSuccessDialog(
+  props: CreateClientSuccessDialogProps
+) {
+  const { visible, clientId } = props;
+  const navigate = useNavigate();
+
+  const calloutIntervalRef = useRef<number>();
+  const [isCalloutVisible, setIsCalloutVisible] = useState(false);
+
+  useEffect(() => {
+    if (calloutIntervalRef.current != null) {
```

Comment:
This should be ran in effect cleanup function instead, otherwise as it is now, it will run just once on component mount.

Reply:
missed `return`, thanks